atomic_reactor/plugins/pre_add_image_content_manifest.py

```diff
@@ -0,0 +1,200 @@
+          "rhel-8-for-x86_64-appstream-rpms"
+      ],
+      "image_contents": [
+        {
+          "purl": "pkg:golang/github.com%2Frelease-engineering%2Fretrodep%2Fv2@v2.0.2",
+          "dependencies": [{"purl": "pkg:golang/github.com%2Fop%2Fgo-logging@v0.0.0"}],
+          "sources": [{"purl": "pkg:golang/github.com%2FMasterminds%2Fsemver@v1.4.2"}]
+        }
+      ]
+    }
+    """
+    key = PLUGIN_ADD_IMAGE_CONTENT_MANIFEST
+    is_allowed_to_fail = False
+    minimal_icm = {
+        'metadata': {
+            'icm_version': 1,
+            'icm_spec': ('https://raw.githubusercontent.com/containerbuildsystem/atomic-reactor/'
+                         'master/atomic_reactor/schemas/content_manifest.json'),
+            'image_layer_index': -1,
+        },
+        'content_sets': [],
+        'image_contents': [],
+    }
+
+    def __init__(self, tasker, workflow, remote_source_icm_url=None, destdir=IMAGE_BUILD_INFO_DIR):
+        """
+        :param tasker: ContainerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param icm_url: str, URL of the ICM from the Cachito request.
+        :param destdir: image path to carry content_manifests data dir
+        """
+        super(AddImageContentManifestPlugin, self).__init__(tasker, workflow)
+        self.content_manifests_dir = os.path.join(destdir, 'content_manifests')
+        self.icm_url = remote_source_icm_url
+        self.dfp = df_parser(self.workflow.builder.df_path, workflow=self.workflow)
+        labels = Labels(self.dfp.labels)
+        _, image_name = labels.get_name_and_value(Labels.LABEL_TYPE_COMPONENT)
+        _, image_version = labels.get_name_and_value(Labels.LABEL_TYPE_VERSION)
+        _, image_release = labels.get_name_and_value(Labels.LABEL_TYPE_RELEASE)
+        self.icm_file_name = '{}-{}-{}.json'.format(image_name, image_version, image_release)
+        self.content_sets = []
+        self._layer_index = None
+        self._icm = None
+
+    @property
+    def layer_index(self):
+        if not self._layer_index:
+            # Default layer index is 1, because base and 'FROM scratch' images
+            #     *always* have 2 layers
+            self._layer_index = 1
+        if not base_image_is_scratch(self.dfp.baseimage):
+            inspect = self.workflow.builder.base_image_inspect
+            self._layer_index = len(inspect[INSPECT_ROOTFS][INSPECT_ROOTFS_LAYERS])
+        return self._layer_index
+
+    @property
+    def icm(self):
+        """
+        Get and validate the ICM from the Cachito API `content-manifest` endpoint.
+
+        :return: dict, the ICM as a Python dict
+        """
+        if self.icm_url is None:
+            self._icm = self.minimal_icm
+        if self._icm is None:
+            session = get_retrying_requests_session()
+
+            # Gets the value of Cachito's 'insecure' key from the active reactor config map
+            session.verify = get_cachito(self.workflow).get('insecure', False)
```

Comment:
If insecure is `True`, should it mean we do **not** want to verify the certificates? In other words, should we toggle whatever the actual insecure value is?

Also, should we default to having a secure connection (i.e., verify if `insecure` is not available)?

Finally, what if, for some reason, cachito configuration is not available? I understand that the resolve_remote_source would skip execution and we would not enter this block. But maybe we should not rely on another plugin to avoid failure here and also set `verify` to `True` in this case, just to be safe.

Reply:
- Absolutely correct; the logic here is wrong (reversed)
- Yes, agreed to default to secure (`insecure = False`)
- I'll make and use a `cachito_verify` instance attribute which defaults to True